std/typecons.d

```diff
@@ -10786,18 +10786,18 @@ private template replaceTypeInFunctionTypeUnless(alias pred, From, To, fun)
         {
             if (i)
                 result ~= ", ";
+            if (storageClasses[i] & ParameterStorageClass.return_)
+                result ~= "return ";
             if (storageClasses[i] & ParameterStorageClass.scope_)
                 result ~= "scope ";
```

Comment:
Why put `return` under the `scope` instead of `ref`? That looks like a breaking change as `return ref scope` will now be translated to `return scope` by `ReplaceTypeUnless`. Although, it's not ideal either way when `ParameterStorageClass` doesn't know about the storage class order like the compiler does.

Reply:
The design of this function is deficient because one cannot control the order of ref, return and scope. I did the best I could with it to keep it working with the unittests.

Reply:
`return ref` and `return scope` should be distinct parameter storage classes, like in the compiler.